### PR TITLE
Add logical not operator using bang !

### LINKED ISSUE #745: KCL doesn't support negation
The AST supports negation/not (the ! operator) but the tokenizer doesn't. So if you try to negate an expression, the program won't tokenize.

Tackle this _after_ #716 is done.

### PR DESCRIPTION
**Pull request by jtran**
Resolves #745.

The motivation for this is that we'd like to be able to use `assert()` easier. For example:

```kcl
assert(!isOutsideRange, "Expected parameters to be inside range 1 to 100")
```

We are conservative in that `!` only works on boolean values. It's currently a parse or runtime error to apply it to any other type of value.

Checklist

- [x] Interpreter
- [x] Codemirror syntax highlighting
- [x] VS Code syntax highlighting?
- [x] Review feedback
- [x] Resolve failing tests by omitting the indexes from the assertions

### CODE REVIEW

**Comment by jessfraz** [comment]
vscode will be handled by semantic tokens :)

File: src/wasm-lib/kcl/src/executor.rs [hunk 1]
```diff
@@ -2710,6 +2719,44 @@ const bracket = startSketchOn('XY')
...
+const t = true
+const f = false
+let notTrue = !t
+let notFalse = !f
+let c = !!true
+let d = !returnTrue()
+// Yup, this is null.
+let myNull = 0 / 0
+let notNull = !myNull
+let notZero = !0
```
Comment (jessfraz) [inline src/wasm-lib/kcl/src/executor.rs:2735]: theres the new assert functions to you could use and parse and execute as tests :)

File: src/wasm-lib/kcl/src/ast/types.rs [hunk 2]
```diff
@@ -3195,6 +3195,18 @@ pub fn parse_json_value_as_string(j: &serde_json::Value) -> Option<String> {
     }
 }
 
+/// Convert JSON value to bool.  Everything is true except for null and false.
+pub fn json_to_bool(j: &serde_json::Value) -> bool {
```
Comment (adamchalmers) [inline src/wasm-lib/kcl/src/ast/types.rs:3199]: I would rather this fail if used with anything other than bool. We can always relax it later. But I think the JS and Python style "truthiness" stuff will be confusing.

**Comment by jtran** [comment]
@jessfraz, the LSP semantic token type **indexes** changed. Is this a real problem, or is it okay to simply update tests to use the new indexes? I haven't yet dug into what they're used for.

**Comment by jtran** [comment]
Jess said it's safe to update the indexes. In that case, I would like to omit the indexes from the test assertions to prevent anyone from running into this again in the future.

**Pull request merged**